formbricks/formbricks
# fix: sync call gets called by other functions

## What does this PR do?



fix: sync call got called by other client functions. What looked like a good approach to save duplicated code first, led to two serverless functions running at the same time (as the executing function needs to wait for the sync function to finish).
With this PR I revamped the sync endpoint, made it easier to follow as first the person is validated and then the state.
All the logic is also wrapped into a helper function to make it usable in the other functions.



## Type of change




## How should this be tested?



- Review the code and spot errors in the logic depending on which attributes are present
- Test the endpoints using the Formbricks demo app locally. Make sure every function gets executed correctly and the desired action is executed in Formbricks app. (test for: sync, set attribute, set userId)
